scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja

```diff
@@ -1182,8 +1182,9 @@ psa_status_t psa_driver_wrapper_cipher_decrypt_setup(
                 operation->id = PSA_CRYPTO_MBED_TLS_DRIVER_ID;
 
             return( status );
-#endif /* MBEDTLS_PSA_BUILTIN_CIPHER */
+#else /* MBEDTLS_PSA_BUILTIN_CIPHER */
             return( PSA_ERROR_NOT_SUPPORTED );
+#endif /* MBEDTLS_PSA_BUILTIN_CIPHER */
```

Comment:
Annoyingly, coming back round to this, I wonder if we would have been better off doing

```
diff --git a/scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja b/scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja
index e716e409f..fe5e469fa 100644
--- a/scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja
+++ b/scripts/data_files/driver_templates/psa_crypto_driver_wrappers.c.jinja
@@ -1181,7 +1181,8 @@ psa_status_t psa_driver_wrapper_cipher_decrypt_setup(
             if( status == PSA_SUCCESS )
                 operation->id = PSA_CRYPTO_MBED_TLS_DRIVER_ID;
 
-            return( status );
+            if( status != PSA_ERROR_NOT_SUPPORTED )
+                return( status );
 #endif /* MBEDTLS_PSA_BUILTIN_CIPHER */
             return( PSA_ERROR_NOT_SUPPORTED );
 
```

which is the existing pattern in the code, around lines 1113, 2062, 2133 and 2205

Reply:
This code will be rewritten when we progress on the generation script. So if it's not breaking anything now, we can wait until it gets rewritten naturally.